include/GafferScene/SceneAlgo.h

```diff
@@ -221,6 +222,24 @@ struct AttributeHistory : public History
 /// null is returned.
 GAFFERSCENE_API AttributeHistory::Ptr attributeHistory( const History *attributesHistory, const IECore::InternedString &attribute );
 
+/// Extends History to provide information on the history of a specific option.
+struct OptionHistory : public History
+{
+	IE_CORE_DECLAREMEMBERPTR( OptionHistory )
+	OptionHistory(
+		const ScenePlugPtr &scene, const Gaffer::ContextPtr &context,
+		const IECore::InternedString &optionName, const IECore::ConstObjectPtr &optionValue
+	) :	History( scene, context ), optionName( optionName ), optionValue( optionValue ) {}
+	IECore::InternedString optionName;
+	IECore::ConstObjectPtr optionValue;
+};
+
+/// Filters `optionsHistory` and returns a history for the specific `option`.
```

Comment:
Nitpick : I think naming the argument `globalsHistory` might be a bit more self-explanatory than `optionsHistory`.